### PR TITLE
Adding support for IParenthesizedOperation to control flow graph

### PR DESCRIPTION
**Pull request by chborl**
@AlekseyTs , @mavasani , @333fred  for review

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParenthesized.cs [hunk 1]
```diff
@@ -496,5 +496,258 @@ static int M1()
 
             VerifyOperationTreeAndDiagnosticsForTest<ReturnStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
+
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParenthesized.cs]: There are no IParenthesizedOperations in any of these tests, so they don't tell us anything about that operation. Please update the tests to have scenarios that actually generate IParenthesizedOperation nodes.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParenthesized.cs]: > There are no IParenthesizedOperations in any of these tests

At the moment C# erases them completely. I explicitly asked to add these tests anyway. At some point we were trying to synthesize the nodes for C# as well, but that became to risky and expensive at the time. We might still have issue to revisit that. 

---
In reply to: [187684261](https://github.com/dotnet/roslyn/pull/26801#discussion_r187684261) [](ancestors = 187684261)

**Comment by chborl** [comment]
@dotnet-bot retest windows_debug_vs-integration_prtest please

File: src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParenthesized.cs [hunk 2]
```diff
@@ -496,5 +496,258 @@ static int M1()
...
+Block[B0] - Entry
+    Statements (0)
+    Next (Regular) Block[B1]
+Block[B1] - Block
+    Predecessors: [B0]
+    Statements (2)
+        IFlowCaptureOperation: 0 (OperationKind.FlowCapture, Type: null, IsImplicit) (Syntax: 'M2')
+          Value: 
+            IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: C, IsImplicit) (Syntax: 'M2')
+
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParenthesized.cs]: Why is the syntax M2 here and not c?

Comment (jaredpar) [reply src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IParenthesized.cs]: Ah nevermind. I see this is capturing the implicit this, not c.

---
In reply to: [187718068](https://github.com/dotnet/roslyn/pull/26801#discussion_r187718068) [](ancestors = 187718068)

**Pull request merged**